### PR TITLE
Update sposet input style for testset.

### PR DESCRIPTION
**Pull request by Hyeondeok-Shin**
## Proposed changes

The most of input styles in testsets are currently using older template which generates `sposet` inside `determinantset`.
This PR is to update its style since older one will be deprecated in the future.
Still some tests using older input styple exist (eg. periodic LCAO, converter, etc.), but the updates for those will be followed up.

## What type(s) of changes does this code introduce?

- Other (please describe): Add test

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?

Workstation

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- No. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

File: tests/estimator/acforce/vmc.fast.in.xml [hunk 1]
```diff
@@ -496,6 +496,8 @@
   1.25722600000000e+00  0.00000000000000e+00  0.00000000000000e+00  0.00000000000000e+00
 </coefficient>
       </sposet>
+    </sposet_builder>
+    <determinantset>
       <slaterdeterminant>
         <determinant id="det_up" sposet="spo-up" size="5"/>
         <determinant id="det_down" sposet="spo-dn" size="4"/>
```
Comment (ye-luo) [inline tests/estimator/acforce/vmc.fast.in.xml:503]: Please remove `id`, `group` and `size` in `determinant`. Also replace `sposet_builder` with `sposet_collection`.

File: tests/io/restart_batch/qmc_short_batch.restart.xml [hunk 2]
```diff
@@ -39,16 +39,15 @@
...
-                  <occupation mode="ground" spindataset="0"/>
-               </determinant>
-               <determinant id="downdet" size="4">
-                  <occupation mode="ground" spindataset="0"/>
-               </determinant>
+               <determinant id="updet" group="u" sposet="spo_ud" size="4"/>
+               <determinant id="downdet" group="d" sposet="spo_ud" size="4"/>
             </slaterdeterminant>
-         </determinantset>
+         </determinantset>
```
Comment (ye-luo) [inline tests/io/restart_batch/qmc_short_batch.restart.xml:50]: Please remove trailing white spaces.

File: tests/molecules/H2O_dimer_sep_pp/H2O.wfs.xml [hunk 3]
```diff
@@ -237,9 +236,9 @@
   2.54780000000000e-02  0.00000000000000e+00  0.00000000000000e+00  3.28770000000000e-02
   0.00000000000000e+00  0.00000000000000e+00  0.00000000000000e+00  0.00000000000000e+00
   0.00000000000000e+00 -5.33000000000000e-03 -3.45700000000000e-03  0.00000000000000e+00
-          </coefficient>
-        </determinant>
-        <determinant id="downdet" size="8">
+	  </coefficient>
```
Comment (ye-luo) [inline tests/molecules/H2O_dimer_sep_pp/H2O.wfs.xml:239]: It seems that a tab was introduced.

File: tests/molecules/LiH_ae_gms/vmc_short_noj.in.xml [hunk 4]
```diff
@@ -6112,8 +6112,8 @@ PROPERTY VALUES FOR THE ROHF  SELF-CONSISTENT FIELD WAVEFUNCTION
 2.950800000000e-02 2.950800000000e-02 0.000000000000e+00 0.000000000000e+00 
 0.000000000000e+00
                   </coefficient>
-               </determinant>
-               <determinant id="downdet" size="2">
+               </sposet>
+               <sposet name="spo-dn" size="2">
```
Comment (ye-luo) [inline tests/molecules/LiH_ae_gms/vmc_short_noj.in.xml:6116]: Strange indents

File: tests/estimator/acforce/vmc.fast.in.xml (hunk 1 above)
Comment (prckent) [reply tests/estimator/acforce/vmc.fast.in.xml:503]: Wondering if we catch the case where the user specifies a particleset and a determinant with different sizes?

Comment (ye-luo) [reply tests/estimator/acforce/vmc.fast.in.xml:503]: determinant size is derived from particleset. `size` in the old input was for constructing sposet only.

File: tests/estimator/sofk/pbyp_dat-h5.xml [hunk 5]
```diff
@@ -38,16 +38,15 @@
...
-                  <occupation mode="ground" spindataset="0"/>
-               </determinant>
-               <determinant id="downdet" size="1">
-                  <occupation mode="ground" spindataset="0"/>
-               </determinant>
+               <determinant id="updet" group="u" sposet="spo_ud" size="1"/>
+               <determinant id="downdet" group="d" sposet="spo_ud" size="1"/>
             </slaterdeterminant>
-         </determinantset>
+         </determinantset>
```
Comment (ye-luo) [inline tests/estimator/sofk/pbyp_dat-h5.xml:49]: trailing white space.

File: tests/estimator/acforce/vmc.fast.in.xml (hunk 1 above)
Comment (prckent) [reply tests/estimator/acforce/vmc.fast.in.xml:503]: Sure, but we should catch inconsistencies in the input. Do we read size anymore? Ideally we would warn the user about unneeded/unused inputs.

Comment (prckent) [reply tests/estimator/acforce/vmc.fast.in.xml:503]: (Not something to fix here)

Comment (ye-luo) [reply tests/estimator/acforce/vmc.fast.in.xml:503]: Once we disallow(warning->error) users creating sposet in determinants. I will issue a warning about all the outdated input.

**Review by prckent — CHANGES_REQUESTED** [review]
I ran all the tests with a modest timeout. After filtering for timed out runs, the list of failures is given below. 
The inputs for `short-H2-orb-opt-16-1`, `short-H4-cjs-16-1`, and `short-H4-orb-opt-dmc-16-1` are broken with parser errors.

```
grep "^[0-9].* Test  #.*.\*" ctest.out |grep -v Timeout|grep -v -- -time
1478/2601 Test  #571: short-LiH_ae-vmc_hf_noj_estimator_energydensity_voronoi-4-4 ..............................................***Failed  Required regular expression not found. Regex=[QMCPACK execution completed successfully
1479/2601 Test  #573: long-LiH_ae-vmc_hf_noj_estimator_energydensity_voronoi-4-4 ...............................................***Failed  Required regular expression not found. Regex=[QMCPACK execution completed successfully
1494/2601 Test  #683: short-LiH_pp-vmc_hf_sdj_estimator_spindensity-4-4 ........................................................***Failed  Required regular expression not found. Regex=[QMCPACK execution completed successfully
1495/2601 Test  #685: long-LiH_pp-vmc_hf_sdj_estimator_spindensity-4-4 .........................................................***Failed  Required regular expression not found. Regex=[QMCPACK execution completed successfully
1497/2601 Test  #691: short-H2-orb-opt-16-1 ....................................................................................***Failed  Error regular expression found in output. Regex=[ERROR]  0.91 sec
1507/2601 Test  #711: short-H4-orb-opt-dmc-16-1 ................................................................................***Failed  Error regular expression found in output. Regex=[ERROR]  5.76 sec
1509/2601 Test  #715: short-H4-cjs-16-1 ........................................................................................***Failed  Error regular expression found in output. Regex=[ERROR]  0.78 sec
1907/2601 Test  #594: long-O_ae_uhf_pyscf-vmc_hf_noj-1-16-samples ..............................................................***Failed    0.04 sec
1926/2601 Test  #619: short-LiH_pp-vmc_hf_sdj_xml-16-1-totenergy ...............................................................***Failed    0.05 sec
1936/2601 Test  #574: long-LiH_ae-vmc_hf_noj_estimator_energydensity_voronoi-4-4-check .........................................***Failed    0.24 sec
1945/2601 Test  #572: short-LiH_ae-vmc_hf_noj_estimator_energydensity_voronoi-4-4-check ........................................***Failed    0.28 sec
1960/2601 Test  #692: short-H2-orb-opt-16-1-totenergy ..........................................................................***Failed    0.04 sec
1967/2601 Test  #712: short-H4-orb-opt-dmc-16-1-totenergy ......................................................................***Failed    0.05 sec
1970/2601 Test  #706: short-H4-RHF-orb-opt-16-1-totenergy ......................................................................***Failed    0.07 sec
1972/2601 Test  #716: short-H4-cjs-16-1-totenergy ..............................................................................***Failed    0.05 sec
1973/2601 Test  #710: short-H4-orb-opt-1-16-totenergy ..........................................................................***Failed    0.07 sec
1976/2601 Test  #722: short-H4-target-descent-16-1-totenergy ...................................................................***Failed    0.05 sec
1979/2601 Test  #730: short-H4-batched-hybrid-16-1-totenergy ...................................................................***Failed    0.04 sec
1980/2601 Test  #732: short-H4-batched-hybrid-1-16-totenergy ...................................................................***Failed    0.05 sec
1984/2601 Test  #686: long-LiH_pp-vmc_hf_sdj_estimator_spindensity-4-4-check ...................................................***Failed    0.17 sec
1992/2601 Test  #749: short-H4-msd-orb-opt-16-1-totenergy ......................................................................***Failed    0.07 sec
1995/2601 Test  #684: short-LiH_pp-vmc_hf_sdj_estimator_spindensity-4-4-check ..................................................***Failed    0.22 sec
1996/2601 Test  #688: short-LiH_pp-vmc_hf_sdj_estimator_energydensity_voronoi-4-4-check ........................................***Failed    0.22 sec
2020/2601 Test  #690: long-LiH_pp-vmc_hf_sdj_estimator_energydensity_voronoi-4-4-check .........................................***Failed    0.33 sec
2043/2601 Test  #852: short-LiH_ae-vmc_msdj_optCI-1-16-variance ................................................................***Failed    0.06 sec
```

**Comment by prckent** [comment]
Thanks Hyeondeok. Fixed the obvious parse errors:
```
 ctest -j 48 -R 'short-H2-orb-opt|short-H4-cjs|short-H4-orb-opt' --output-on-failure 
...
79% tests passed, 3 tests failed out of 14

Label Time Summary:
QMCPACK                     = 14254.79 sec*proc (7 tests)
QMCPACK-checking-results    =   0.23 sec*proc (7 tests)
abort                       = 13925.79 sec*proc (10 tests)
bug                         = 13925.79 sec*proc (10 tests)
quality_unknown             = 14255.01 sec*proc (14 tests)
unstable                    =   0.23 sec*proc (7 tests)

Total Test time (real) = 762.07 sec

The following tests FAILED:
        694 - short-H4-orb-opt-1-16-totenergy (Failed)
        695 - short-H4-orb-opt-dmc-16-1 (Failed)
        696 - short-H4-orb-opt-dmc-16-1-totenergy (Failed)
Errors while running CTest
```
The orb-opt-dmc test fails with a bad DMC run. Unfortunately that is "normal" and consistent with before.

**Review by prckent — APPROVED** [review]

**Comment by prckent** [comment]
Test this please

**Comment by Hyeondeok-Shin** [comment]
@prckent Yes, I'm getting same failure with older input style. It seems need to be revisited later. I'll take a look more when I work for remaining style update.

**Pull request merged**